gatsby-browser.js

```diff
@@ -6,4 +6,24 @@
 
 import wrapPageElement from './gatsby/wrap-page-element';
 
-export { wrapPageElement };
+const onRouteUpdate = ({ location }) => {
+  if (process.env.NODE_ENV !== `production` || typeof ga !== `function`) {
+    return null;
+  }
+
+  // wrap inside a timeout to make sure react-helmet is done with it's changes (https://github.com/gatsbyjs/gatsby/issues/9139)
+  // reactHelmet is using requestAnimationFrame: https://github.com/nfl/react-helmet/blob/5.2.0/src/HelmetUtils.js#L296-L299
+  const sendPageView = () => {
+    const pagePath = location
+      ? location.pathname + location.search + location.hash
+      : undefined;
+    window.ga(`set`, `page`, pagePath);
+    window.ga(`send`, `pageview`);
+  };
+
+  // Minimum delay for reactHelmet's requestAnimationFrame
```

Comment:
Would you be willing to explain this a little more. I'm unclear about why this necessary.

Reply:
apparently it's a common issue that google analytics will fire the pageview off before helmet has finished injecting the page titles, so folks were seeing improperly tracked pageviews. this is gatsby's current answer for dealing with that